src/main/java/com/amazonaws/services/sqs/SQSScheduledExecutorService.java

```diff
@@ -122,8 +122,8 @@ public int compareTo(Delayed other) {
         public SendMessageRequest toSendMessageRequest() {
             SendMessageRequest request = super.toSendMessageRequest();
 
-            int sqsDelaySeconds = (int)Math.min(TimeUnit.NANOSECONDS.toSeconds(delay), MAX_SQS_DELAY_SECONDS);
-            if (sqsDelaySeconds < 0) {
+            int sqsDelaySeconds = (int)Math.min(TimeUnit.NANOSECONDS.toSeconds(Math.abs(delay)), MAX_SQS_DELAY_SECONDS);
```

Comment:
nit: How about use `Math.max(1, sqsDelaySeconds)`?

Reply:
I like that, thanks!